src/chainstate/stacks/boot/pox.clar

```diff
@@ -257,36 +279,63 @@
-                { pox-addr: pox-addr, first-reward-cycle: first-reward-cycle, num-cycles: num-cycles, amount-ustx: amount-ustx, i: u0 })))
-        (err ERR_STACKING_UNREACHABLE))
-
-    ;; mark address in use over this range
-    (map-set pox-addr-reward-cycles
-        { pox-addr: pox-addr }
-        { first-reward-cycle: first-reward-cycle, num-cycles: num-cycles }
-    )
-    (ok true)
-    )
-)
+     (is-eq num-cycles 
+            (get i (fold add-pox-addr-to-ith-reward-cycle cycle-indexes 
+                         { pox-addr: pox-addr, first-reward-cycle: first-reward-cycle, num-cycles: num-cycles, amount-ustx: amount-ustx, i: u0 })))
+     (err ERR_STACKING_UNREACHABLE))
+    (ok true)))
+
+(define-private (add-pox-partial-stacked-to-ith-cycle
+                 (cycle-index uint)
+                 (params { pox-addr: { version: (buff 1), hashbytes: (buff 20) },
+                           reward-cycle: uint,
+                           num-cycles: uint,
+                           amount-ustx: uint }))
+  (let ((pox-addr     (get pox-addr     params))
+        (num-cycles   (get num-cycles   params))
+        (reward-cycle (get reward-cycle params))
+        (amount-ustx  (get amount-ustx  params)))
+    (let ((current-amount
+           (default-to u0
+             (get stacked-amount
+                  (map-get? partial-stacked-by-cycle { sender: tx-sender, pox-addr: pox-addr, reward-cycle: reward-cycle })))))
+      (if (>= cycle-index num-cycles)
+          ;; do not add to cycles >= cycle-index
+          false
+          ;; otherwise, add to the partial-stacked-by-cycle
+          (map-set partial-stacked-by-cycle
+                   { sender: tx-sender, pox-addr: pox-addr, reward-cycle: reward-cycle }
+                   { stacked-amount: (+ amount-ustx current-amount) }))
+      ;; produce the next params tuple
+      { pox-addr: pox-addr,
+        reward-cycle: (+ u1 reward-cycle),
+        num-cycles: num-cycles,
+        amount-ustx: amount-ustx })))
+
+;; Add a PoX address to a given sequence of partial reward cycle lists.
+;; A PoX address can be added to at most 12 consecutive cycles.
+;; No checking is done.
+(define-private (add-pox-partial-stacked (pox-addr (tuple (version (buff 1)) (hashbytes (buff 20))))
+                                         (first-reward-cycle uint)
+                                         (num-cycles uint)
+                                         (amount-ustx uint))
+  (let ((cycle-indexes (list u0 u1 u2 u3 u4 u5 u6 u7 u8 u9 u10 u11)))
+    (fold add-pox-partial-stacked-to-ith-cycle cycle-indexes 
+          { pox-addr: pox-addr, reward-cycle: first-reward-cycle, num-cycles: num-cycles, amount-ustx: amount-ustx })
+    true))
 
 ;; What is the minimum number of uSTX to be stacked in the given reward cycle?
 ;; Used internally by the Stacks node, and visible publicly.
 (define-read-only (get-stacking-minimum)
-    (/ stx-liquid-supply u20000)
-)
+    (/ stx-liquid-supply STACKING_THRESHOLD_25))
```

Comment:
So, the threshold won't be dynamic in this PR?  Can you open an issue to make it dynamic?

Reply:
That's correct, opened #1947